kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ExpressionUtils.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.expressions;
+
+import java.math.BigDecimal;
+import java.util.Comparator;
+
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.types.*;
+
+import static io.delta.kernel.defaults.internal.DefaultKernelUtils.checkArgument;
+
+class ExpressionUtils {
+    private ExpressionUtils() {}
+
+    /**
+     * Utility method that calculates the nullability result from given two vectors. Result is
+     * null if at least one side is a null.
+     */
+    static boolean[] nullabilityEval(ColumnVector left, ColumnVector right) {
+        int numRows = left.getSize();
+        boolean[] nullability = new boolean[numRows];
+        for (int rowId = 0; rowId < numRows; rowId++) {
+            nullability[rowId] = left.isNullAt(rowId) || right.isNullAt(rowId);
+        }
+        return nullability;
+    }
+
+    /**
+     * Utility method to compare the left and right according to the natural ordering
+     * and return the comparison result (-1, 0, 1) for each row as an integer array.
+     */
+    static int[] compareVectors(ColumnVector left, ColumnVector right) {
```

Comment:
nit: Vectors might be superfluous. params are obviously vectors. but now the name is inconsistent with other types expressions like the `nullabilityEval` that do not end with "Vectors" but does operate on Vectors exactly like this compare function.

Reply:
renamed this to `compare` and the other one to `evalNullability`.